apps/examples/board_specific_demo/ameba_wifi_csi_demo/example_wifi_csi.c

```diff
@@ -0,0 +1,128 @@
+/******************************************************************************
+ *
+ * Copyright(c) 2007 - 2015 Realtek Corporation. All rights reserved.
+ *
+ *
+ ******************************************************************************/
+ /******************************************************************************
+
+ * Copyright 2022 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+
+  ******************************************************************************/
+
+
+#include <stdio.h>
+#include <tinyara/config.h>
+#include <tinyara/wifi/rtk/wifi_structures.h>
+#include <example_wifi_csi.h>
+#include <sched.h>
+
+#define IP_ADDR_INVALID "0.0.0.0"
+extern int wifi_csi_config(rtw_csi_action_parm_t *act_param);
+extern int wifi_csi_report(uint32_t buf_len, uint8_t *csi_buf, uint32_t *len, rtw_csi_header_t *csi_header);
+extern int wifi_is_running(unsigned char wlan_idx);
+static sem_t wc_ready_sema;
+
+unsigned int csi_raw_data_len = 2048;
+
+/* wifi csi report callback */
+void example_wifi_csi_report_cb(char *buf, int buf_len, int flags, void *userdata)
+{
+	rtw_up_sema(&wc_ready_sema);
+	csi_raw_data_len = flags;
+	return;
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int wificsi_main(int argc, char *argv[])
+#endif
+{
+	printf("WIFI CSI\n");
+	rtw_csi_action_parm_t act_param = {0};
+	rtw_csi_header_t csi_hdr = {0};
+	unsigned int len;
+	unsigned char *csi_buf = NULL;
+	act_param.group_num = 0;
+	act_param.mode = 2;  /* currently only supported */
+	act_param.accuracy = 0;
+	act_param.trig_period = 200;  /* ms */
+	act_param.data_rate = 0xC;  /* ofdm 6 mpbs*/
+//	act_param.mac_addr = {0x00, 0xe0, 0x4c, 0x81, 0x92, 0xbb};
+	char ipv4_address[4];
+	char ipv4_buf[16];
+	while (1) {
+		if (netlib_get_ipv4addr("wlan0", ipv4_address) == 0) {
+			snprintf(ipv4_buf, 16, "%d.%d.%d.%d", ipv4_address[0], ipv4_address[1], ipv4_address[2], ipv4_address[3]);
+		}
+		if (wifi_is_running(WLAN0_IDX) && ((wifi_get_join_status() == RTW_JOINSTATUS_SUCCESS) && (strncmp(ipv4_buf, IP_ADDR_INVALID, sizeof(ipv4_buf))))) {
+			sleep(2); /* 2s */
+			break;
+		}
+		sleep(2); /* 2s */
+	}
+	/**
+	 * should use semaphore to wait wifi event happen
+	 * the following example shows that we wait for wifi csi ready
+	 */
+	rtw_init_sema(&wc_ready_sema, 0);

```

Comment:
Let's use POSIX sem functions, sem_init, sem_wait and sem_post which are defined in <semaphore.h>

Reply:
same as RTW_FALSE -> false and RTW_TRUE -> true
They are defined in <stdbool.h>

Reply:
Dear @sunghan-chang,
I have changed all to POSIX sem functions.